src/chunk.c

```diff
@@ -3909,6 +3954,22 @@ find_hypertable_from_table_or_cagg(Cache *hcache, Oid relid, bool allow_matht)
 	return ht;
 }
 
+Datum
+ts_chunk_freeze_chunk(PG_FUNCTION_ARGS)
+{
+	Oid chunk_relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+	TS_PREVENT_FUNC_IF_READ_ONLY();
+	Chunk *chunk = ts_chunk_get_by_relid(chunk_relid, true);
+	if (chunk->relkind != RELKIND_RELATION)
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("operation not supported on %s", get_rel_name(chunk_relid))));
```

Comment:
I guess this error could happen for distributed hypertable, right? Maybe we should explicitly mention that.

Also this API likely can be supported in the same way as `compress_chunk`, which calls compress function remotely for foreign relations

Reply:
added more detail to error message.